src/software/thunderscope/binary_context_managers/tigers_autoref.py

```
@@ -0,0 +1,233 @@
+
+import logging
+import os
+import threading
+import time
+
+
+class TigersAutoref(object):
+    """
+    A wrapper over the TigersAutoref binary. It coordinates communication between the Simulator, TigersAutoref and
+    Gamecontroller. 
+
+    In CI mode, the flow of data corresponds to:
+
+              SSL_DetectionFrame          AutoRefCiOutput: TrackerWrapperPacket                CiOutput
+    Simulator ------------------> AutoRef ------------------------------------> Gamecontroller --------------> AI
+                AutoRefCiInput                         CiInput                                 RefereeMessage
+
+    For more information: https://github.com/RoboCup-SSL/ssl-game-controller/blob/master/doc/AutoRefCi.md
+    """
+
+    AUTOREF_COMM_PORT = 10013
+    AUTOREF_NUM_RETRIES = 20
+    NEXT_PACKET_DELAY = 1.0 / 30  # 30 Hz
+
+    def __init__(
+        self,
+        tick_rate_ms: int,
+        ci_mode: bool = False,
+        autoref_runtime_dir: str = None,
+        buffer_size: int = 5,
+        gc: Gamecontroller = Gamecontroller(),
+        supress_logs: bool = True,
+        show_gui: bool = False,
+    ) -> None:
+        self.tigers_autoref_proc = None
+        self.auto_ref_proc_thread = None
+        self.auto_ref_wrapper_thread = None
+        self.ci_mode = ci_mode
+        self.autoref_runtime_dir = autoref_runtime_dir
+        self.wrapper_buffer = ThreadSafeBuffer(buffer_size, SSL_WrapperPacket)
+        self.referee_buffer = ThreadSafeBuffer(buffer_size, Referee)
+        self.gamecontroller = gc
+        self.supress_logs = supress_logs
+        self.tick_rate_ms = tick_rate_ms
+        self.current_timestamp = int(time.time_ns())
+        self.show_gui = show_gui
+
+    def __enter__(self) -> "self":
+        if not os.path.exists("/opt/tbotspython/autoReferee/bin/autoReferee"):
+            logging.info(
+                "Could not find autoref binary, did you run ./setup_software.sh"
+            )
+            return
```

We should probably throw instead here, since we're not setting up the autoref. Or we should bump the log msg to a warning.

---

bumped to a warning